elastic/kibana
# [7.17][Security Solution][Alerts] Add empty meta object in filters

## Summary

https://github.com/elastic/kibana/issues/126574

The code here was refactored significantly in 8.0, but the bug with missing the `meta` field in `filters` remained. This PR allows alerts with `filters` that don't have `meta` populated to be sent to the timeline.

Testing:
1. Create documents to trigger the rule. Adjust the `@timestamp` and `event.ingested` values to current time and make a second document with a different `source.geo.country_name` value.
```
PUT auditbeat-1
{
  "mappings": {
    "properties": {
      "@timestamp": {
        "type": "date"
      },
      "event.category": {
        "type": "keyword"
      },
      "event.ingested": {
        "type": "date"
      },
      "event.outcome": {
        "type": "keyword"
      },
      "source.ip": {
        "type": "keyword"
      },
      "source.geo.country_name": {
        "type": "keyword"
      },
      "user.name": {
        "type": "keyword"
      },
      "host.name": {
        "type": "keyword"
      },
      "otherField": {
        "type": "keyword"
      }
    }
  }
}

POST auditbeat-1/_doc
{
  "@timestamp": 1649109892000,
  "event.ingested": 1649109892000,
  "user.name": "you",
  "event.category": "authentication",
  "event.outcome": "success",
  "source.ip": "test",
  "source.geo.country_name": "usa"
}
```
2. Create a threshold rule with a filter that doesn't have a `meta` field. Easiest way is to import the rules file [here](https://gist.github.com/marshallmain/0c24f772c9be212b29da423ddbc7c999). The rule will run on import and should create an alert.
3. Click `Investigate in timeline` on the alert. It should successfully open the timeline.
